ACE/ace/SSL/SSL_Context.cpp

```diff
@@ -274,39 +254,8 @@ ACE_SSL_Context::set_mode (int mode)
     case ACE_SSL_Context::SSLv23:
       method = ::SSLv23_method ();
       break;
-    case ACE_SSL_Context::TLSv1_client:
-      method = ::TLSv1_client_method ();
-      break;
-    case ACE_SSL_Context::TLSv1_server:
-      method = ::TLSv1_server_method ();
-      break;
-    case ACE_SSL_Context::TLSv1:
-      method = ::TLSv1_method ();
-      break;
-#if defined(TLS1_1_VERSION) && (TLS_MAX_VERSION >= TLS1_1_VERSION)
-    case ACE_SSL_Context::TLSv1_1_client:
-      method = ::TLSv1_1_client_method ();
-      break;
-    case ACE_SSL_Context::TLSv1_1_server:
-      method = ::TLSv1_1_server_method ();
-      break;
-    case ACE_SSL_Context::TLSv1_1:
-      method = ::TLSv1_1_method ();
-      break;
-#endif
-#if defined(TLS1_2_VERSION) && (TLS_MAX_VERSION >= TLS1_2_VERSION)
-    case ACE_SSL_Context::TLSv1_2_client:
-      method = ::TLSv1_2_client_method ();
-      break;
-    case ACE_SSL_Context::TLSv1_2_server:
-      method = ::TLSv1_2_server_method ();
-      break;
-    case ACE_SSL_Context::TLSv1_2:
-      method = ::TLSv1_2_method ();
-      break;
-#endif
     default:
-      method = ::SSLv3_method ();
+      method = ::SSLv23_method ();
```

Comment:
You seem to be misunderstanding something, but I'm not sure what.

First, that part of the documentation is removed because the SSLv2
support has been removed.  That was a section about using the
SSLv2 compatible client hello.  You really don't want that since
it doesn't support important extentions, some of which were always
enabled.  So it was actually impossible to send such a client
hello in the first place.  The 0.9.8 version was the last one that
actually send such SSLv2 compatible client hellos.  It didn't even
support TLS 1.1 and it support will end the end of this month.

Second, As it point out, it will support multiple versions of the
protocol.  It's the only option that supports multiple protocols.
Like it also point out is that if you want to restrict the
supported protocols you should tell it to use the options.
There is actually an upcomming new API that lets you set a minimum
version instead of using all those settings.

This is a part that you might want to add at some point, to be
able to disable SSLv3 for instance.  That would have as effect
that if the other side only speaks SSLv3 that you refuse to talk
to it.

Third, I don't recommend using the ciphers to select a protocol,
it's not really meant for that.

Fouth, by default you will get TLS 1.2 if both side support it
instead of the insecure SSLv3.  You now get the strong security
by default.  If TLS 1.3 support is added you will also
automatically upgraded to it.